### PR TITLE
Dasd 7576 cloud service migration

### PR DESCRIPTION
**Pull request by cpwevans**

### CODE REVIEW

File: src/api/SFA.DAS.Provider.Events.Api/NLog.config [hunk 1]
```diff
@@ -1,45 +1,16 @@
...
   </targets>
 
   <rules>
-    <!-- add your logging rules here -->
-    <logger name="*" minlevel="Trace" writeTo="redis" />
-    <!--
-    Write all events with minimal level of Debug (So Debug, Info, Warn, Error and Fatal, but not Trace)  to "f"
-    <logger name="*" minlevel="Debug" writeTo="f" />
-    -->
+    <logger name="*" minlevel="Debug" writeTo="Redis" />
```
Comment (nbowes24) [inline src/api/SFA.DAS.Provider.Events.Api/NLog.config:14]: Debug level isn't suitable for releases

Comment (jack-cloud-platform) [reply src/api/SFA.DAS.Provider.Events.Api/NLog.config:14]: Currently Debug in the das-dotnet-template, e.g. https://github.com/SkillsFundingAgency/das-dotnet-templates/blob/bd0a77544d4ddeedd716eebbe2cc0bcf1378b604/src/working/templates/Web/src/SFA.DAS.WebTemplateSourceName.Web/nlog.config#L18 so resolving for now.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  value: SFA.DAS.Provider.Events.Api
+- name: BuildConfiguration
+  value: release
+- name: BuildPlatform
+  value: any cpu
+- group: RELEASE Management Resources
+- group: RELEASE das-providerevents
+
+pool:
+  vmImage: 'vs2017-win2016'
```
[the remark this answers is not in the document]
Comment (jack-cloud-platform) [reply azure-pipelines.yml]: Updated to own agents

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  - template: pipeline-templates/job/deploy.yml
+    parameters:
+      Environment: TEST2
+      ServiceConnection: SFA-DAS-DevTest-ARM
+
+- stage: Deploy_PP
+  dependsOn: Build
+  displayName: Deploy to PP
+  variables:
+  - group: PreProd Management Resources
```
[the remark this answers is not in the document]
Comment (jack-cloud-platform) [reply azure-pipelines.yml]: Corrected

File: src/api/SFA.DAS.Provider.Events.Api/NLog.config [hunk 4]
```diff
@@ -1,45 +1,16 @@
...
-      <field name="requestId" layout="${aspnet-traceidentifier}"/>
-      <field name="loggerTitle" layout="${logger}"/>
-    </target>
-
-    <!--
-    Write events to a file with the date in the filename.
-    <target xsi:type="File" name="f" fileName="${basedir}/logs/${shortdate}.log"
-            layout="${longdate} ${uppercase:${level}} ${message}" />
-    -->
+    <target xsi:type="Redis" name="Redis" connectionStringName="LoggingConnectionString" environmentKeyName="EnvironmentName" appName="das-providerevents-api" includeAllProperties="true" layout="${message}"></target>
```
Comment (hardik-desai-kcl) [inline src/api/SFA.DAS.Provider.Events.Api/NLog.config:10]: App insights Target is missing

<target xsi:type="ApplicationInsights" name="ApplicationInsights" InstrumentationKey="<KeyHere>" environmentKeyName="EnvironmentName" appName="das-providerevents-api" includeAllProperties="true" layout="${message}"></target>

File: src/collections/README.md [hunk 5]
```diff
@@ -1,6 +0,0 @@
-# Collections Earnings (BETA)
```
Comment (hardik-desai-kcl) [inline src/collections/README.md:1]: do we need to keep a simple README.md even if its empty

File: src/api/SFA.DAS.Provider.Events.Api/Web.config.orig [hunk 6]
```diff
@@ -9,7 +9,6 @@
   </configSections>
   <appSettings>
     <add key="LogLevel" value="Debug"/>
-    <add key="LoggingConnectionString" value="server=.;database=PaymentsEventApi;trusted_connection=true;" />
 <<<<<<< HEAD
```
Comment (hardik-desai-kcl) [inline src/api/SFA.DAS.Provider.Events.Api/Web.config.orig:12]: this file should not be checked in as it gets created when there is a merge conflict, please delete this file

File: src/api/SFA.DAS.Provider.Events.sln [hunk 7]
```diff
@@ -27,8 +27,6 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SFA.DAS.Provider.Events.Api
 EndProject
```
Comment (hardik-desai-kcl) [inline src/api/SFA.DAS.Provider.Events.sln:27]: remove this section as both of these files have been deleted

Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Build", "Build", "{DAFC5414-B8A0-433A-B50E-367087ABD671}"
	ProjectSection(SolutionItems) = preProject
		Build.fsx = Build.fsx
		RunBuild.bat = RunBuild.bat
	EndProjectSection
EndProject

**Review by hardik-desai-kcl — CHANGES_REQUESTED** [review]

File: src/api/SFA.DAS.Provider.Events.Api/NLog.config (hunk 4 above)
Comment (jack-cloud-platform) [reply src/api/SFA.DAS.Provider.Events.Api/NLog.config:10]: As discussed, no app insights target initially, however we do have transaction search visibility from out of the box app insights logging.

File: src/api/SFA.DAS.Provider.Events.Api/Web.config.orig (hunk 6 above)
Comment (jack-cloud-platform) [reply src/api/SFA.DAS.Provider.Events.Api/Web.config.orig]: File removed

File: src/api/SFA.DAS.Provider.Events.sln (hunk 7 above)
Comment (jack-cloud-platform) [reply src/api/SFA.DAS.Provider.Events.sln:21]: Section removed

File: src/collections/README.md (hunk 5 above)
Comment (jack-cloud-platform) [reply src/collections/README.md]: Removal reverted

**Review by hardik-desai-kcl — APPROVED** [review]

**Pull request merged**